sale_blanket_order/wizard/create_sale_orders.py

```diff
@@ -0,0 +1,201 @@
+                raise UserError(
+                    _("Sale Blanket Order %s is not open") % line.order_id.name
+                )
+            line_company_id = line.company_id and line.company_id.id or False
+            if company_id is not False and line_company_id != company_id:
+                raise UserError(_("You have to select lines " "from the same company."))
+            else:
+                company_id = line_company_id
+
+    @api.model
+    def _default_lines(self):
+        blanket_order_line_obj = self.env["sale.blanket.order.line"]
+        blanket_order_line_ids = self.env.context.get("active_ids", False)
+        active_model = self.env.context.get("active_model", False)
+
+        if active_model == "sale.blanket.order":
+            bo_lines = self._default_order().line_ids
+        else:
+            bo_lines = blanket_order_line_obj.browse(blanket_order_line_ids)
+
+        self._check_valid_blanket_order_line(bo_lines)
+
+        lines = [
+            (
+                0,
+                0,
+                {
+                    "blanket_line_id": bo_line.id,
+                    "product_id": bo_line.product_id.id,
+                    "date_schedule": bo_line.date_schedule,
+                    "remaining_uom_qty": bo_line.remaining_uom_qty,
+                    "price_unit": bo_line.price_unit,
+                    "product_uom": bo_line.product_uom,
+                    "qty": bo_line.remaining_uom_qty,
+                    "partner_id": bo_line.partner_id,
+                },
+            )
+            for bo_line in bo_lines.filtered(lambda l: l.remaining_uom_qty != 0.0)
+        ]
+        return lines
+
+    blanket_order_id = fields.Many2one(
+        comodel_name="sale.blanket.order",
+        readonly=True,
+        default=lambda self: self._default_order(),
+    )
+    sale_order_id = fields.Many2one(
+        "sale.order", string="Purchase Order", domain=[("state", "=", "draft")]
+    )
+    line_ids = fields.One2many(
+        "sale.blanket.order.wizard.line",
+        "wizard_id",
+        string="Lines",
+        default=_default_lines,
+    )
+
+    def _prepare_so_line_vals(self, line):
+        return {
+            "product_id": line.product_id.id,
+            "name": line.product_id.name,
+            "product_uom": line.product_uom.id,
+            "sequence": line.blanket_line_id.sequence,
+            "price_unit": line.blanket_line_id.price_unit,
+            "blanket_order_line": line.blanket_line_id.id,
+            "product_uom_qty": line.qty,
+            "tax_id": [(6, 0, line.taxes_id.ids)],
+            "analytic_tag_ids": [(6, 0, line.blanket_line_id.analytic_tag_ids.ids)],
+        }
+
+    def create_sale_order(self):
+        order_lines_by_customer = defaultdict(list)
+        currency_id = 0
+        pricelist_id = 0
+        user_id = 0
+        payment_term_id = 0
+        for line in self.line_ids.filtered(lambda l: l.qty != 0.0):
```

Comment:
I am getting empty lines here.

Reply:
Hi, I'm migrating this to 15 and I got the line.remaining_uom_qty equals zero to here. Any ideas how this could happen?